### PR TITLE
SBS-3638: Add customer-specified survey coverage

### PR DESCRIPTION
**Pull request by audunska**
Input welcome

### CODE REVIEW

File: cognite/seismic/protos/types.proto [hunk 1]
```diff
@@ -213,6 +214,18 @@ message Geometry {
...
 
+// User-provided coverage override for surveys
+message SurveyCoverageOverride {
+    message NoOverride { };
+    oneof override {
+        // Overrides survey coverage with the provided geometry
+        Geometry coverage_override = 1;
+        // Specifies that coverage should not be overridden, but computed from
+        // the seismicstores in the survey
+        NoOverride no_override = 2;
```
Comment (BenChand) [inline cognite/seismic/protos/types.proto:225]: Will this be what it defaults to anyway if survey coverage override is none?

Comment (audunska) [reply cognite/seismic/protos/types.proto:225]: Yes, but it is useful to have a separate variant for this, so we can edit it away.

An EditSurveyRequest with `coverage_override = None` is interpreted as "do nothing, leave it as it is". So in order to remove the override, we need to set it explicitly as `NoOverride`.

Cf. `SurveyGridTransformation`, where the `DeduceFromTraces` variant plays the same role.

Comment (BenChand) [reply cognite/seismic/protos/types.proto:225]: That makes sense, thanks

**Review by BenChand — APPROVED** [review]
LGTM

**Comment by audunska** [comment]
I'm not sure what to do about the `crs` field of the `proto::Geometry` argument. Do we ignore it and assume the geometry is in the `crs` of the survey? Do we accept it and try to convert the geometry from that to `survey`'s crs?

**Pull request merged**